stripe/src/main/java/com/stripe/android/view/CountryAdapter.java

```diff
@@ -23,24 +22,60 @@
 /**
  * Adapter that populates a list of countries for a spinner.
  */
-class CountryAdapter extends BaseAdapter implements SpinnerAdapter {
+class CountryAdapter extends ArrayAdapter {
+
+    @VisibleForTesting
+    List<String> mCountries;
+    @VisibleForTesting
+    List<String> mSuggestions;
+    private Filter mFilter;
 
-    private List<Pair<String, String>> mCountryToDisplayName;
     private Context mContext;
 
-    public CountryAdapter(Context context) {
+    CountryAdapter(Context context, List<String> countries) {
+        super(context, R.layout.menu_text_view);
         mContext = context;
-        mCountryToDisplayName = getOrderedCountries();
+        mCountries = getOrderedCountries(countries);
+        mSuggestions = mCountries;
+        mFilter = new Filter() {
+            @Override
+            protected FilterResults performFiltering(CharSequence charSequence) {
+                FilterResults filterResults = new FilterResults();
+                List<String> suggestedCountries = new ArrayList<>();
+                if (charSequence == null) {
+                    filterResults.values = mCountries;
+                    return filterResults;
+                }
+                for (String country : mCountries) {
+                    if (country.toLowerCase().startsWith(charSequence.toString().toLowerCase())) {
```

Comment:
Ordinarily, I'd be a little bit scared of doing toLowerCase in a loop (you could sort of optimize by taking the toLowerCase for the input value out of the loop), but since this is performed off the main thread it's probably fine, compared to the overhead of keeping a parallel list of countries all in lowercase (or making an enum out of the countries).

Actually, now that I type it out, minor nit: definitely do the `charSequence.toString().toLowerCase()` outside the for loop :)